code/game/machinery/computer/arcade.dm

```diff
@@ -189,107 +247,301 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
+			player_mp += 10
+			arcade_action(usr,"defend",attackamt)
+			playsound(src, 'sound/arcade/mana.ogg', 50, TRUE, extrarange = -3)
+
+		//mainly used to counter short temper and their absurd damage, will deal twice the damage the player took of a non magical attack.
+		else if(href_list["counter_attack"] && player_mp >= 10)
+			temp = "<br><center><h3>you prepare yourself to counter the next attack!</h3></center>"
+			player_mp -= 10
+			arcade_action(usr,"counter_attack",attackamt)
+			playsound(src, 'sound/arcade/mana.ogg', 50, TRUE, extrarange = -3)
+
+		else if(href_list["counter_attack"] && player_mp < 10)
+			temp = "<br><center><h3>you don't have the mp necessary to counter attack and defend yourself instead</h3></center>"
+			player_mp += 10
+			arcade_action(usr,"defend",attackamt)
+			playsound(src, 'sound/arcade/mana.ogg', 50, TRUE, extrarange = -3)
+
+		//power attack deals twice the amount of damage but is really expensive MP wise, mainly used with combos to get weakpoints.
+		else if (href_list["power_attack"] && player_mp >= 20)
+			temp = "<br><center><h3>You attack [enemy_name] with all your might for [attackamt * 2] damage!</h3></center>"
+			enemy_hp -= attackamt * 2
+			player_mp -= 20
+			arcade_action(usr,"power_attack",attackamt)
+
+		else if(href_list["power_attack"] && player_mp < 20)
+			temp = "<br><center><h3>You don't have the mp necessary for a power attack and settle for a light attack!</h3></center>"
+			enemy_hp -= attackamt
+			arcade_action(usr,"attack",attackamt)
 
 	if (href_list["close"])
 		usr.unset_machine()
 		usr << browse(null, "window=arcade")
 
 	else if (href_list["newgame"]) //Reset everything
-		temp = "New Round"
-		player_hp = 30
-		player_mp = 10
-		enemy_hp = 45
-		enemy_mp = 20
-		gameover = FALSE
-		turtle = 0
+		temp = "<br><center><h3>New Round<center><h3>"
 
 		if(obj_flags & EMAGGED)
 			Reset()
 			obj_flags &= ~EMAGGED
 
+		enemy_setup(gamerSkill)
+		screen_setup(usr)
+
+
 	add_fingerprint(usr)
 	updateUsrDialog()
 	return
 
-/obj/machinery/computer/arcade/battle/proc/arcade_action(mob/user)
+
+///happens after a player action and before the enemy turn. the enemy turn will be cancelled if there's a gameover.
+/obj/machinery/computer/arcade/battle/proc/arcade_action(mob/user,player_stance,attackamt)
+	screen_setup(user)
+	blocked = TRUE
+	if(player_stance == "attack" || player_stance == "power_attack")
+		if(attackamt > 40)
+			playsound(src, 'sound/arcade/boom.ogg', 50, TRUE, extrarange = -3)
+		else
+			playsound(src, 'sound/arcade/hit.ogg', 50, TRUE, extrarange = -3)
+
+	timer_id = addtimer(CALLBACK(src, .proc/enemy_action,player_stance,user),1 SECONDS,TIMER_STOPPABLE)
+	gameover_check(user)
+
+
+///the enemy turn, the enemy's action entirely depend on their current passive and a teensy tiny bit of randomness
+/obj/machinery/computer/arcade/battle/proc/enemy_action(player_stance,mob/user)
+	temp = ""
```

Comment:
Store this info as a list and at the end of it do `list.Join()`, possibly just `temp = list_temp.Join()`
It's much faster to do list operations than string ones here.